miniaudio.py

```diff
@@ -1152,42 +1152,69 @@ def _internal_data_callback(device: ffi.CData, output: ffi.CData, input: ffi.CDa
     callback_device = _callback_data[userdata_id]  # type: Union[PlaybackDevice, CaptureDevice, DuplexStream]
     callback_device._data_callback(device, output, input, framecount)
 
+@ffi.def_extern()
+def _internal_stop_callback(device: ffi.CData) -> None:
+    if not device.pUserData:
+        return
+    userdata_id = struct.unpack('q', ffi.unpack(ffi.cast("char *", device.pUserData), struct.calcsize('q')))[0]
+    callback_device = _callback_data[userdata_id]  # type: Union[PlaybackDevice, CaptureDevice, DuplexStream]
+    callback_device._stop_callback(device)
+
+PlaybackCallbackGeneratorType = Generator[Union[bytes, array.array], int, None]
+CaptureCallbackGeneratorType = Generator[None, Union[bytes, array.array], None]
+DuplexCallbackGeneratorType = Generator[Union[bytes, array.array], Union[bytes, array.array], None]
+GeneratorTypes = Union[PlaybackCallbackGeneratorType, CaptureCallbackGeneratorType, DuplexCallbackGeneratorType]
+
 
 class AbstractDevice:
     def __init__(self):
+        self.running = False # type: Boolean
```

Comment:
that type annotation should not be necessary?

Reply:
I'm not too familiar with typing hints in python, so I may have this wrong. 
To my understanding, this helps other parts know to expect `self.running` should be a bool (previously I used the incorrect type of Boolean). 
Is this not the case?

Reply:
The type analizer will see that is a boolean automatically because you assign False

Reply:
Ah! Thanks for the clarification. Will remove the comment.